File: rootfs/etc/nginx/lua/nginx/ngx_srv_redirect.lua
local request_uri = ngx.var.request_uri
local redirect_to = ngx.arg[1]

local luaconfig = ngx.shared.luaconfig
local use_forwarded_headers = luaconfig:get("use_forwarded_headers")


if string.sub(request_uri, -1) == "/" then
    request_uri = string.sub(request_uri, 1, -2)
end

local redirectScheme = ngx.var.scheme
local redirectPort = ngx.var.server_port

if use_forwarded_headers then
    if ngx.var.http_x_forwarded_proto then
        redirectScheme = ngx.var.http_x_forwarded_proto
    end
    if ngx.var.http_x_forwarded_port then
        redirectPort = ngx.var.http_x_forwarded_port
    end
end


return string.format("%s://%s:%s%s", redirectScheme,
    redirect_to, redirectPort, request_uri)

File: rootfs/etc/nginx/js/nginx/ngx_srv_redirect.js
function srv_redirect(req) {
    const redirectTo = req.variables.tmp_redirect_to;

    const requestUri = req.variables.request_uri.replace(/\/$/, '');

    const useForwardedHeaders = req.variables.forwarded_headers
    const xForwardedProto = req.variables.http_x_forwarded_proto;
    const xForwardedPort = req.variables.http_x_forwarded_port;

    const redirectScheme = useForwardedHeaders && xForwardedProto ? xForwardedProto : req.variables.scheme;
    const redirectPort = useForwardedHeaders && xForwardedPort ? xForwardedPort : req.variables.server_port;

    return `${redirectScheme}://${redirectTo}:${redirectPort}${requestUri}`;
}

export default { srv_redirect };
